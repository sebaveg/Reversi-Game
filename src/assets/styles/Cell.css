.cell {
  border: 1px solid white;
  color: rgba(255, 255, 255, .2);
  cursor: pointer;
  height: 70px;
  width: 70px;
}
.cell:hover {
  color: white;
}
.highlight {
  border: 1px solid yellow;
  color: yellow;
  cursor: pointer;
  height: 70px;
  width: 70px;
  cursor: cell;
}
.highlight:hover {
  border: 1px solid gold;
  color: gold;
}
.black, .white {
  height: 55px;
  width: 55px;
  border: 2px solid grey;
  border-radius: 50%;
  margin: 0 auto;
  cursor: default;
}

.black {
  background-color: black;
}

.white {
  background-color: white;
}

@media (max-width: 1020px) {
  .cell {
    height: 60px;
    width: 60px;
  }

  .highlight {
    height: 60px;
    width: 60px;
  }

  .black,
  .white {
    height: 45px;
    width: 45px;
  }
}

@media (max-width: 720px) {
  .cell {
    height: 50px;
    width: 50px;
  }
  .highlight {
    height: 50px;
    width: 50px;
  }
  .black,
  .white {
    height: 35px;
    width: 35px;
  }
}

@media (max-width: 520px) {
  .cell {
    height: 35px;
    width: 35px;
  }
  .highlight {
    height: 35px;
    width: 35px;
  }
  .black,
  .white {
    height: 25px;
    width: 25px;
  }
}