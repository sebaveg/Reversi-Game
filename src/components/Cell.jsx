/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React from 'react';

const Cell = ({
  disk, allowed, children, onClick,
}) => (
  <td className={allowed.length > 0 ? 'highlight' : 'cell'} onClick={onClick}>
    {disk !== 'black' && disk !== 'white' ? children : null}
    <div className={disk} />
  </td>
);

export default Cell;
