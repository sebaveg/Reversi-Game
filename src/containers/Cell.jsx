import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  setError,
  setBoard,
  changeTurn,
  putDisks,
} from '../actions';

import CellLayout from '../components/Cell';

import '../assets/styles/Cell.css';

class Cell extends Component {
  diskOponent = () => (this.props.currentPlayer === 'white' ? 'black' : 'white')

  inBoard = (x, y) => (x >= 0) && (x < 8) && (y >= 0) && (y < 8)

  // reverse disks oponent
  async reverse() {
    if (this.props.allowed.length > 0) {
      const { board } = this.props;
      const x = this.props.position[0];
      const y = this.props.position[1];
      // await this.props.putDisks({
      //   x, y, disk: this.props.currentPlayer, allowedCells: this.props.board[x][y].allowedCell,
      // });
      // board[x][y].disk = this.props.currentPlayer;
      board[x][y].allowedCell.forEach((cell) => {
        this.props.putDisks({ x: cell.X, y: cell.Y, disk: this.props.currentPlayer });
      });
      // this.props.setBoard(board);
      await this.props.changeTurn();
    } else {
      this.props.setError('No can put your disk here');
      setTimeout(() => {
        this.props.setError('');
      }, 2000);
    }
  }

  render() {
    return (
      <CellLayout
        disk={this.props.disk}
        allowed={this.props.allowed}
        onClick={this.reverse.bind(this)}
      >
        {this.props.children}
      </CellLayout>
    );
  }
}

const mapStateToProps = (state) => ({
  board: state.board.present.board,
  currentPlayer: state.players.present.currentPlayer,
});

const mapDispatchToProps = {
  setError,
  setBoard,
  changeTurn,
  putDisks,
};

export default connect(mapStateToProps, mapDispatchToProps)(Cell);
