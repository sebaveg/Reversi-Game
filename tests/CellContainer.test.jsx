/* eslint-disable no-undef */
import React from 'react';
import configureMockStore from 'redux-mock-store';

import Cell from '../src/containers/Cell';

import CellLayout from '../src/components/Cell';

const initialState = {
  error: '',
  board: {
    past: [],
    present: {
      board: [
        [{ id: 'A1', allowedCell: [{ X: 1, Y: 1 }] }, { id: 'A2' }, { id: 'A3' }, { id: 'A4' }, { id: 'A5' }, { id: 'A6' }, { id: 'A7' }, { id: 'A8' }],
        [{ id: 'B1' }, { id: 'B2', allowedCell: [{ X: 2, Y: 2 }] }, { id: 'B3' }, { id: 'B4' }, { id: 'B5' }, { id: 'B6' }, { id: 'B7' }, { id: 'B8' }],
        [{ id: 'C1' }, { id: 'C2' }, { id: 'C3' }, { id: 'C4' }, { id: 'C5' }, { id: 'C6' }, { id: 'C7' }, { id: 'C8' }],
        [{ id: 'D1' }, { id: 'D2' }, { id: 'D3' }, { id: 'D4' }, { id: 'D5' }, { id: 'D6' }, { id: 'D7' }, { id: 'D8' }],
        [{ id: 'E1' }, { id: 'E2' }, { id: 'E3' }, { id: 'E4' }, { id: 'E5' }, { id: 'E6' }, { id: 'E7' }, { id: 'E8' }],
        [{ id: 'F1' }, { id: 'F2' }, { id: 'F3' }, { id: 'F4' }, { id: 'F5' }, { id: 'F6' }, { id: 'F7' }, { id: 'F8' }],
        [{ id: 'G1' }, { id: 'G2' }, { id: 'G3' }, { id: 'G4' }, { id: 'G5' }, { id: 'G6' }, { id: 'G7' }, { id: 'G8' }],
        [{ id: 'H1' }, { id: 'H2' }, { id: 'H3' }, { id: 'H4' }, { id: 'H5' }, { id: 'H6' }, { id: 'H7' }, { id: 'H8' }],
      ],
      currentPlayer: 'black', // Array 2D
    },
    future: [],
  },
};

const mockStore = configureMockStore();
const store = mockStore(initialState);

const setUp = (props = {}) => {
  // eslint-disable-next-line react/jsx-props-no-spreading
  const component = shallow(<Cell store={store} {...props} />).childAt(0).dive();
  return component;
};

describe('<Cell /> Container', () => {
  let wrapper;
  describe('<Cell /> Container without allowed cells', () => {
    const props = {
      disk: 'white',
      allowed: [],
      position: [0, 0],
      onClick: jest.fn(),
    };
    beforeEach(() => {
      wrapper = setUp(props);
    });

    it(' +++ capturing snapshot of home', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render one <CellLayout /> component', () => {
      const component = wrapper.find(CellLayout);
      expect(component.length).toBe(1);
    });

    it('Props in <CellLayout /> component', () => {
      const props = {
        disk: 'white',
        allowed: [[0, 0]],
        onClick: jest.fn(),
      };
      const component = wrapper.find(CellLayout);
      expect(component.props().disk).toBe(props.disk);
      // expect(component.props().allowed).toBe(props.allowed);
    });

    it('Event onClick <CellLayout /> dispatches the correct actions', () => {
      wrapper.instance().reverse();
      const actions = store.getActions();
      expect(actions).toEqual([{
        type: 'set_error',
        payload: 'No can put your disk here',
      }]);
      // expect(actions.setError.mock.calls.length).toBe(0);
      // expect(props.setPosDisksWhite).toHaveBeenCalled();
      // expect(actions.setPosDisksWhite.mock.calls.length).toBe(0);
    });
  });

  describe('<Cell /> Container With allowed cells Disk WHITE', () => {
    const props = {
      disk: 'white',
      allowed: [[1, 1]],
      position: [1, 1],
      onClick: jest.fn(),
    };
    beforeEach(() => {
      wrapper = setUp(props);
    });
    it('Event onClick <CellLayout /> dispatches the correct actions', () => {
      wrapper.instance().reverse();
      const actions = store.getActions();
      expect(actions).toEqual([
        {
          payload: 'No can put your disk here',
          type: 'set_error',
        }, {
          payload: {
            disk: 'black',
            x: 2,
            y: 2,
          },
          type: 'put_disks',
        }, {
          payload: [1, 1],
          type: 'set_position_disk_black',
        }, {
          payload: {
            color: 'black',
            pos: [1, 1],
          },
          type: 'set_position_disk',
        }, {
          type: 'change_turn',
        },
      ]);
    });
  });

  describe('<Cell /> Container With allowed cells Disk BLACK', () => {
    const props = {
      disk: 'black',
      allowed: [[0, 0]],
      position: [0, 0],
      onClick: jest.fn(),
    };
    beforeEach(() => {
      wrapper = setUp(props);
    });
    it('Event onClick <CellLayout /> dispatches the correct actions', () => {
      wrapper.instance().reverse();
      const actions = store.getActions();
      expect(actions).toEqual([
        {
          type: 'set_error',
          payload: 'No can put your disk here',
        },
        {
          payload: {
            disk: 'black',
            x: 2,
            y: 2,
          },
          type: 'put_disks',
        },
        {
          payload: [1, 1],
          type: 'set_position_disk_black',
        },
        {
          payload: {
            color: 'black',
            pos: [1, 1],
          },
          type: 'set_position_disk',
        },
        {
          type: 'change_turn',
        },
        {
          payload: {
            disk: 'black',
            x: 1,
            y: 1,
          },
          type: 'put_disks',
        },
        {
          payload: [0, 0],
          type: 'set_position_disk_black',
        },
        {
          payload: {
            color: 'black',
            pos: [0, 0],
          },
          type: 'set_position_disk',
        },
        {
          type: 'change_turn',
        },

      ]);
    });
  });
});
