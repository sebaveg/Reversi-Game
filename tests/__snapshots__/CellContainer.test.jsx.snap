// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Container  +++ capturing snapshot of home 1`] = `
<Cell
  onClick={[Function]}
/>
`;
